edb/edgeql/compiler/viewgen.py

```diff
@@ -446,6 +473,133 @@ def gen_pointers_from_defaults(
     return [result[i] for i in topological.sort(graph, allow_unresolved=True)]
 
 
+def _apply_rewrites(
+    pointers: Dict[s_pointers.Pointer, Optional[irast.Set]],
+    specified_ptrs: Set[sn.UnqualName],
+    rewrite_kind: qltypes.RewriteKind,
+    view_scls: s_objtypes.ObjectType,
+    ir_set: irast.Set,
+    stype: s_objtypes.ObjectType,
+    path_id_namespace: Optional[irast.Namespace],
+    view_rptr: Optional[context.ViewRPtr],
+    exprtype: s_types.ExprType,
+    ctx: context.ContextLevel,
+) -> List[Tuple[s_pointers.Pointer, irast.Set | None]]:
+    result: List[Tuple[s_pointers.Pointer, irast.Set | None]] = []
+
+    path_id = ir_set.path_id
+    scls_pointers = stype.get_pointers(ctx.env.schema)
+
+    # init set that will be used for __subject__
+    subject_path = irast.PathId.from_type(
+        ctx.env.schema,
+        stype,
+        typename=sn.QualName(module="__derived__", name="__subject__"),
+    )
+    subject_set = irast.Set(
```

Comment:
Don't create the `Set` explicitly, go through `new_set`

Reply:
And `ensure_set`/`expression_set` for when you have an Expr you want to turn into a `Set`. It will generate new pathids for it, too, if you don't give one